### PR TITLE
WIP: Rewrite classes to es6 syntax

### PR DESCRIPTION
**Pull request by mathe42**
This is a WIP rewrite to use the "new" es6 class syntax.

Goal is to intodruce no (or as little as possible) breaking changes. I try to keep the diff as small as posible for this PR so near 0 code changes. 

I create this PR now so there can be some feedback (by the leaflet-Team) about the change. I will stop working on this until some Team members had a look on this PR!

## Why?
ES6 was released in 2015 so 6 years ago. Since then the JS syntax got a loot more features that makes code easier and simpler to read. So I think it is a good thing if you use "modern" JS-Syntax.

## What is bad about the leaflet class strukture?
Nothing. 
But there are some advantages about the new syntax:

1. Typescript support can be added with simple JSDoc / TSDoc comments + maybe 1-2 .d.ts files.
2. IDE can infer a loot with the new syntax.
3. Better DX for new developers.

## What about old Browsers?
I transpile the code with esbuild back to es6 syntax and use `buble` or `babel` to create es5 syntax. This is important to remove the new class syntax and check if the prototype chain and all works properly.

I currently create 3 Output files:

* Modern (es2017 wich is supported by 95% of Browser user)
* es6 wich is support by many old versions of Browsers.
* legacy wich supports IE (currently not shure wich versions)

My plan is that the legacy build supports all browsers that leaflet currently supports. Output files have will be renamed so that there is no breaking change when finishing this PR.

But if a developer don't need IE support they can use the modern or es6 build. Or transpile the es2017 build to some other target.

Note to support IE <= 9 we need to transpile to es3 see https://caniuse.com/es5. Will propably use https://github.com/google/closure-compiler to do that.

## What about Plugins?
Plugins can use the old class syntax or can use the es6 class syntax. If they get transpiled to es5 / es3 there should be no problems to support even old IE. If useing es6 class syntax out of the box intellisense without adding `@types/leaflet`

## Current breaking changes
List of breaking changes. all of them I will try to fix. 

* Docs?
* Events Mixin is removed
* initHooks in parent class get called
* dist file names

## Summary
1. No breaking changes (if possible)
2. IE support
3. Moders JS sytax (not only classes)
4. Much smaller build if you only support modern browsers.
5. Needed for proper Treeshaking (there are some more problems before this (Mixins) works but this is needed! But that are breaking changes)

## Related
see #7591, #7055

### CODE REVIEW

**Comment by mathe42** [comment]
Disccusion if this is a good efford or not should happen in #7615

File: src/layer/FeatureGroup.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ import {LatLngBounds} from '../geo/LatLngBounds';
  * 	.addTo(map);
  * ```
  */
-
+console.log(LayerGroup);
```
Comment (Falke-Design) [inline src/layer/FeatureGroup.js:25]: console.log(LayerGroup);

Comment (mathe42) [reply src/layer/FeatureGroup.js:25]: Oh Thanks! Will remove that. But as said above still WIP. So there might be some console.log or debugger calls in the code...

**Review by IvanSanchez — CHANGES_REQUESTED** [review]

**Comment by mathe42** [comment]
see #7651 for new pr

**Pull request closed without merging**